exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/LateralJoinBatchCreator.java

```diff
@@ -29,6 +30,12 @@
   @Override
   public LateralJoinBatch getBatch(ExecutorFragmentContext context, LateralJoinPOP config, List<RecordBatch> children)
       throws ExecutionSetupException {
-    return new LateralJoinBatch(config, context, children.get(0), children.get(1));
+    LateralJoinBatch ljBatch = new LateralJoinBatch(config, context, children.get(0), children.get(1));
+    UnnestPOP unnest = config.getUnnestForLateralJoin();
+    if (unnest != null) {
+      unnest.setLateral(ljBatch);
+      unnest.getUnnestBatch().setLateralJoinBatch(ljBatch);
```

Comment:
As discussed we don't need to set Lateral inside UnnestPop anymore since RecordBatch is already created. So just setting inside RecordBatch should be good enough.

But looking more into unnest code it looks like it is using `popConfig.getLateral()` in all the places to communicate with Lateral. So may be we can leave line 36 and remove line 37 and related methods.